### ISSUE TITLE
Boiler temperature fixed on 35 degrees and keeps heating even though the PID is at -300%

### ISSUE THREAD

**Issue by Dezorian** [rsciriano/ESPHome-OpenTherm#35]
The main problem is that my heating is still using gas to heat up the heating water even though the PID is -100% or lower; I see heating going on (no hot water being used). Why is that? I can only stop my house from heating up by turning off the heating PID. Boiler modulation is also not working. It looks like the heating always has a maximum time for being off before starting to heat the water again and therefore still heating up my floor (and temperatures still going up).

I have now set an automation in Home Assistant which set the heating on or off, that shouldn't be the way to do this.

My return temperature is -1 (Intergas HRE 36), but i've read that thats something the boiler doesn't return (so no problem?)

I have two circuits (floor heating and radiator heating). The radiator heating has a seperate wireless thermostat which is set at a very low temperature so it should not interfere with the OpenTherm adapter.

What am I missing? 

In short, this is what happens:
![Screenshot 2022-10-12 152801](https://user-images.githubusercontent.com/7250520/195355689-93cdefcb-2bfb-4f45-b475-1f04a1564a23.png)

**Comment by Dezorian** [comment]
I think someone else has this same problem. It looks like its in the [opentherm library itself](https://github.com/ihormelnyk/opentherm_library/issues/40). Will check this out.

**Comment by foltymat** [comment]
Hi @Dezorian, I'm not quite at the -100% mark just yet but I feel like even at any -x.x% value, the flame should not be on. If you don't mind, what kind of adapter are you using to communicate with the boiler? I bought the one rom DIYLESS but more importantly, are you using a fully fledged OpenTherm gateway, or just the Master shield (that's what I'm trying to do).

**Comment by Dezorian** [comment]
Yeah, I bought the same one with Wemos D1 Mini. So no gateway. Unfortunately I also found out I can only use my OpenTherm connection with my floor heating loop and I'm actually looking for controlling my radiator loop (higher temperature) which can only be controlled wirelessly (Honeywell Wireless Round). So this got a bit of low priority for me.

**Comment by foltymat** [comment]
Understood. One question still came to my mind though. Could you tell me how quickly/often you see changes of the PID ERROR and PID Result parameters? I feel like they are crucial to the operation but no matter what i do they don't change that quickly. I would assume, after changing the Set value, the ERROR parameter would change immediately.

Thanks for the knowledge, by the way 😊

**Comment by straaat** [comment]
The PID loop only runs when the temperature sensor updates. I use a zigbee sensor from HA and add a heartbeat filter for predictable operation.

**Comment by foltymat** [comment]
That actually makes sense. My logic was it should be calculated whenever I changed the Set value. Looks like you are correct!
